### PR TITLE
wrappers/services: RemainAfterExit=yes for oneshot daemons w/ stop cmds

### PR DESCRIPTION
**Pull request by ssweeny**
systemd complains if you generate a service file for a oneshot daemon
that also includes an ExecStop directive but doesn't specify
RemainAfterExit=yes. systemd needs this keep track of the service after
the command exits.

Also adds a test for this and fixes a typo in the opening comment of the generated systemd unit file.

Fixes: https://bugs.launchpad.net/snappy/+bug/1647169

### CODE REVIEW

File: wrappers/services.go [hunk 1]
```diff
@@ -170,7 +170,7 @@ func RemoveSnapServices(s *snap.Info, inter interacter) error {
 
 func genServiceFile(appInfo *snap.AppInfo) string {
 	serviceTemplate := `[Unit]
-# Auto-generated, DO NO EDIT
+# Auto-generated, DO NOT EDIT
```
Comment (zyga) [inline wrappers/services.go]: Nice :-)

File: wrappers/services.go [hunk 2]
```diff
@@ -204,6 +205,17 @@ WantedBy={{.ServicesTarget}}
 		restartCond = systemd.RestartOnFailure.String()
 	}
 
+	var remain string
+	if appInfo.Daemon == "oneshot" {
```
Comment (zyga) [inline wrappers/services.go]: Just thinking aloud. Is there any other condition where we would like to have remain-after-exit set?

File: wrappers/services_gen_test.go [hunk 3]
```diff
@@ -201,3 +202,29 @@ apps:
...
+    app:
+        command: bin/start
+        stop-command: bin/stop
+        reload-command: bin/reload
+        post-stop-command: bin/stop --post
+        stop-timeout: 10s
+        daemon: oneshot
+`
+
+	info, err := snap.InfoFromSnapYaml([]byte(yamlText))
```
Comment (zyga) [inline wrappers/services_gen_test.go]: We have `snaptest.MockYaml` from `snap/snaptest` for this purpose. Please use it here

Comment (ssweeny) [inline wrappers/services_gen_test.go]: @zyga Do you mean `snaptest.MockSnap`? I don't see a `MockYaml` anywhere.

Comment (zyga) [inline wrappers/services_gen_test.go]: Sorry, I meant MockInfo :-) They differ subtly but I think all you need is MockInfo

**Pull request merged**